trie/utils/verkle.go

```diff
@@ -28,7 +28,7 @@ const (
 	CodeHashLeafKey  = 1
 
 	BasicDataVersionOffset  = 0
-	BasicDataCodeSizeOffset = 5
+	BasicDataCodeSizeOffset = 4
```

Comment:
As mentioned in [another PR comment](https://github.com/gballet/go-ethereum/pull/472/files#r1724816418), the idea is that we can write one extra byte to the left (thus why `5`->`4`) since that byte will be zero anyway.

To be honest, I don't personally like doing this for some reasons:
- This isn't aligned with how the spec describes offsets.
- Why 4 and not 3? or 2? etc

I'd prefer using `4` which is what the spec says, and even do a verification that the actual code size is 3 byte max indeed. The way this is coded today seems to imply that a 4 byte encoded code size is fine which it isn't.

Not pushing for changing it, just giving my opinion. If we want to keep it this way, I'd highly recommend at least adding a comment here on why this number isn't `4` as the spec says... other people will wonder the same since it's the first thing they'll check when reviewing this line.